script/shared/agent.ts

```diff
@@ -0,0 +1,147 @@
+import path from "path";
+import fs from "fs";
+
+import { Config } from "./config";
+import { defaultContextPath } from "./constants";
+import { Context, ContextHook } from "./context";
+import { extractByte32AddrFromBech32 } from "./utils";
+import { getContractInfo } from "./wasm";
+
+export type HplAgentConfig = {
+  name: string;
+  domainId: string;
+  chainId: string;
+  protocol: "cosmos";
+  rpcUrls: { http: string }[];
+  grpcUrl: string;
```

Comment:
Ii think this is now also an array of objects?

Reply:
oh i missed that